paasta_tools/vitesscluster_tools.py

```diff
@@ -756,6 +771,27 @@ def get_keyspaces(self) -> List[KeyspaceConfigDict]:
     def get_update_strategy(self) -> Dict[str, str]:
         return {"type": "Immediate"}
 
+    def get_prometheus_pushgateway_config(self) -> PrometheusPushGatewayConfigDict:
+        config = PrometheusPushGatewayConfigDict(
+            name="prometheusPushGateway",
+            image="prometheus/pushgateway",
+            restartPolicy="OnFailure",
+            command=['--web.listen-address=":8000"'],
+        )
+        return config
+
+    def get_load_vitess_metrics_config(self) -> VitessMetricsConfigDict:
+        config = VitessMetricsConfigDict(
+            name="vitessMetricScrapper",
+            restartPolicy="OnFailure",
+            command=[
+                "sh",
+                "-c",
+                "puppet:///modules/profile_vitess/scrape-vitess-containers.sh",
```

Comment:
Wanted to make sure what's proper way of ensuring a specific bash script is mounted within a paasta pod? Will update this value based on feedback.

Reply:
this won't work as-is: paasta is pretty unaware of puppet. there's two ways of using a script in a paasta pod:
1. include it inside your containers
2. puppet the script onto kube nodes and mount the script into the pod

we don't really do per-pool puppet (for reasons™ - i.e., y/lasagna) - so i'd probably recommend including this script in your docker images

Reply:
that said: i wonder if instead of porting over this pushgateway complexity we can instead have a simpler exporter that converts the vitess metrics into prometheus

(i.e., a single sidecar rather than two)

Reply:
we previously used this pattern for uwsgi metrics before centralizing to a single centralized exporter :)

Reply:
to add a file to your image (which is what i'd recommend), you'd want to `COPY` the file from the repo that the image is built from into the image in the dockerfile

if we're mirroring an upstream image here, we'll need to do some additional work to create a dockerfile that starts from the mirrored image and adds any customizations

that said: i think this would be a whole lot simpler if we had a separate image just for something that would re-export the vitess metrics in a prometheus-scrapable format - e.g., we used to run github.com/timonwong/uwsgi_exporter as a sidecar for uwsgi-autoscaled instances to re-export the uwsgi metrics (which are not prometheus-scrapable) in a scrapable format

Reply:
Yeah that seems like a better idea, was mostly just trying to bring parity first to our puppet setup. If it's a requirement we can re-write the bash. Below is the puppet version which we run a cron job, hence why we push metrics through the gateway. Might also give better context of our solution initially at least:
https://sourcegraph.yelpcorp.com/sysgit/puppet@master/-/blob/modules/profile_vitess/manifests/init.pp?L168-186

Reply:
(if PE is fine with using their pushgateway here or y'all are willing to manage your own pushgateway setup): probably the easiest way to do this would be to create an image just for this script and push that to docker-paasta alongside the other vitess images

that said: the script will need a significant bit of refactoring - i think the only reusable bits are the curls to get metrics and the curls to send the metrics to the pushgateway (all the docker stuff will not work inside the sidecar).

Reply:
Ah good catch almost didn't see those, yeah those should be replaceable with pod configs from vitess etc from the config. Will reach out to PE and work on creating a new script

Reply:
@alexanderjulianmartinez IMO, we cannot use this script as is. We'd need to modify the hostname and port with prometheus_pushgateway sidecar pod name and port in https://sourcegraph.yelpcorp.com/sysgit/puppet@4edf3b7b030e04241c495cbe7e8075ad556a6b0f/-/blob/modules/profile_vitess/files/scrape-vitess-containers.sh

And add this shell script to the docker image, vitess_base used by vttablet, vtctld and vtgate https://sourcegraph.yelpcorp.com/docker-images/vitess_base. vtadmin has a separate docker image if we're scraping it as well

Reply:
(unless the sidecar needs binaries and whatnot provided by the vitess base image, it might be easier to have a separate image just for the sidecar: that's what we did when we ran the uwsgi exporter sidecar)

Reply:
(that said, it will be cheaper to run a single sidecar that is prometheus scrapable than to run 2 sidecars for this push workflow)